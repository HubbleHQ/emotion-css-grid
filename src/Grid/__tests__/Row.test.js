import React from 'react';
import { shallow } from 'enzyme';

import { Row, GridConfigProvider } from '..';

const mockContent = <p>Hello World</p>;

describe('Grid : Row', () => {
  it('should render as expected in its default state', () => {
    const output = shallow(
      <GridConfigProvider mobileBreakpoint={600}>
        <Row>{mockContent}</Row>
      </GridConfigProvider>
    );
    expect(output).toMatchSnapshot();
    const children = output.props().children;
    expect(children).toMatchSnapshot();
  });
  it('should render with display:block when "allowMismatchedHeights" is TRUE', () => {
    const output = shallow(
      <GridConfigProvider mobileBreakpoint={600} allowMismatchedHeights>
        <Row>{mockContent}</Row>
      </GridConfigProvider>,
    );
    expect(output).toMatchSnapshot();
    const children = output.props().children;
    expect(children).toMatchSnapshot();
  });
  it('should render with display:flex (on non-mobile screens) when "allowMismatchedHeights" is explicitly FALSE', () => {
    const output = shallow(
      <GridConfigProvider mobileBreakpoint={600} allowMismatchedHeights={false}>
        <Row>{mockContent}</Row>
      </GridConfigProvider>,
    );
    expect(output).toMatchSnapshot();
    const children = output.props().children;
    expect(children).toMatchSnapshot();
  });
});
