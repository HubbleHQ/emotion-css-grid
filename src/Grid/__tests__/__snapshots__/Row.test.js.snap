// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid : Row should render as expected in its default state 1`] = `
<ContextProvider
  value={
    Object {
      "children": <Row
        allowMismatchedHeights={false}
        className=""
      >
        <p>
          Hello World
        </p>
      </Row>,
      "columnNumber": 12,
      "gridMaxWidth": 1920,
      "gutterWidth": 20,
      "mobileBreakpoint": 600,
    }
  }
>
  <Row
    allowMismatchedHeights={false}
    className=""
  >
    <p>
      Hello World
    </p>
  </Row>
</ContextProvider>
`;

exports[`Grid : Row should render as expected in its default state 2`] = `
<Row
  allowMismatchedHeights={false}
  className=""
>
  <p>
    Hello World
  </p>
</Row>
`;

exports[`Grid : Row should render with display:block when "allowMismatchedHeights" is TRUE 1`] = `
<ContextProvider
  value={
    Object {
      "allowMismatchedHeights": true,
      "children": <Row
        allowMismatchedHeights={false}
        className=""
      >
        <p>
          Hello World
        </p>
      </Row>,
      "columnNumber": 12,
      "gridMaxWidth": 1920,
      "gutterWidth": 20,
      "mobileBreakpoint": 600,
    }
  }
>
  <Row
    allowMismatchedHeights={false}
    className=""
  >
    <p>
      Hello World
    </p>
  </Row>
</ContextProvider>
`;

exports[`Grid : Row should render with display:block when "allowMismatchedHeights" is TRUE 2`] = `
<Row
  allowMismatchedHeights={false}
  className=""
>
  <p>
    Hello World
  </p>
</Row>
`;

exports[`Grid : Row should render with display:flex (on non-mobile screens) when "allowMismatchedHeights" is explicitly FALSE 1`] = `
<ContextProvider
  value={
    Object {
      "allowMismatchedHeights": false,
      "children": <Row
        allowMismatchedHeights={false}
        className=""
      >
        <p>
          Hello World
        </p>
      </Row>,
      "columnNumber": 12,
      "gridMaxWidth": 1920,
      "gutterWidth": 20,
      "mobileBreakpoint": 600,
    }
  }
>
  <Row
    allowMismatchedHeights={false}
    className=""
  >
    <p>
      Hello World
    </p>
  </Row>
</ContextProvider>
`;

exports[`Grid : Row should render with display:flex (on non-mobile screens) when "allowMismatchedHeights" is explicitly FALSE 2`] = `
<Row
  allowMismatchedHeights={false}
  className=""
>
  <p>
    Hello World
  </p>
</Row>
`;
